addons/snailmail/static/src/messaging/component/message/message.js

```diff
@@ -0,0 +1,115 @@
+odoo.define('snailmail.messaging.component.Message', function (require) {
+'use strict';
+
+const components = {
+    Message: require('mail.messaging.component.Message'),
+    SnailmailErrorDialog: require('snailmail.messaging.component.SnailmailErrorDialog'),
+};
+
+const { patch } = require('web.utils');
+
+const { useState } = owl;
+
+Object.assign(components.Message.components, {
+    SnailmailErrorDialog: components.SnailmailErrorDialog,
+});
+
+patch(components.Message, 'snailmail.messaging.component.Message', {
+    /**
+     * @override
+     */
+    _constructor() {
+        this._super(...arguments);
+        this.snailmailState = useState({
+            // Determine if the error dialog is displayed.
+            hasDialog: false,
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    async _doActionFormat() {
+        this.env.do_action('snailmail.snailmail_letter_format_error_action', {
+            additional_context: {
+                message_id: this.message.id,
+            },
+        });
+    },
+    /**
+     * @private
+     */
+    async _doActionMissingFields() {
+        const letterIds = await this.env.rpc({
```

Comment:
avoid async behaviour in component, this is a sign this should be on entity

Reply:
This one feels crap even when moved to entity, but I guess that's a first step.